src/js/view/controls/settings-menu.js

```diff
@@ -171,6 +171,12 @@ export function setupSubmenuListeners(settingsMenu, controlbar, viewModel, api)
             controlbar.toggleCaptionsButtonState(!!index);
         }
     }, settingsMenu);
+    model.on('change:playlistItem', () => {
```

Comment:
This shouldn't affect the behavior, but let's put this handler above the `captionsList` handler to be safe since the order of events should be playlistItem > captionsList > captionsIndex.